### PR TITLE
feat(worker): save workers info

### PR DESCRIPTION
**Pull request by roggervalf**

### CODE REVIEW

File: src/classes/queue.ts [hunk 1]
```diff
@@ -167,6 +169,23 @@ export class Queue<
     this.emit('resumed');
   }
 
+  async getWorkersInfo(start = 0, end = 1, asc = false): Promise<string[]> {
+    const client = await this.client;
+
+    // Removes workers info by cleaner timeout
+    await client.zremrangebyscore(
```
[the remark this answers is not in the document]
Comment (manast) [reply src/classes/queue.ts:176]: The question is where to refactor it because both queue and worker will need to use it.

[the remark this answers is not in the document]
Comment (manast) [reply src/classes/queue.ts:176]: specially since we have the ```workerCleanTimeout``` option which should match both the workers and the getter here.

File: src/classes/worker.ts [hunk 2]
```diff
@@ -189,7 +195,20 @@ export class Worker<
             () => v4(),
           );
 
+          let lastWorkerInfoUpdate = Date.now();
+
+          await this.saveWorkerInfo(lastWorkerInfoUpdate);
+
           while (!this.closing) {
+            const currentTime = Date.now();
```
Comment (manast) [inline src/classes/worker.ts:203]: The problem here is that if the job takes more time to process than the "workerCleanTimeout", then the worker will disappear from the workers list.

**Review by manast — COMMENTED** [review]
Please check my comments. I think we need to revisit this functionality. Maybe there is a better way to do this.
Let me give you some hints. Currently a worker needs to have a dedicated connection to redis right?, this is because we need a blocking connection to it. This means that there is a 1:1 mapping between workers and connections.
So we should be able to actually use the "CLIENT SETNAME" command (https://redis.io/commands/client-setname), its just that we need to encode in the name all the information we want to store for the worker. 
Then we need to use https://redis.io/commands/client-list and filter out by workers and only the workers belonging to a given queue.
So basically the same mechanism we have today but instead we store more data on the name, so the question is how to encode the data, considering spaces are not allowed and also that we may want to implement the worker getter as a lua script to filter out before actually returning the data...

**Comment by roggervalf** [comment]
https://github.com/taskforcesh/bullmq/pull/965 should cover same logic as in this pr
